tensorflow/core/util/tensor_bundle/tensor_bundle_test.cc

```diff
@@ -259,6 +272,207 @@ void TestBasic() {
   }
 }
 
+// Reinterpret the bytes of an rvalue
+#define _CAST(x, T) (*reinterpret_cast<const T*>(&x))
+
+// Type-specific subroutine of SwapBytes test below
+template <typename T>
+void TestByteSwap(const T* forward, const T* swapped, int array_len) {
+  auto bytes_per_elem = sizeof(T);
+
+  // Convert the entire array at once
+  std::unique_ptr<T> forward_copy(new T[array_len]);
+  std::memcpy(forward_copy.get(), forward, array_len * bytes_per_elem);
+  TF_EXPECT_OK(
+      ByteSwapArray((char*)forward_copy.get(), bytes_per_elem, array_len));
+  for (int i = 0; i < array_len; i++) {
+    EXPECT_EQ(forward_copy.get()[i], swapped[i]);
+  }
+
+  // Then the array wrapped in a tensor
+  auto shape = TensorShape({array_len});
+  auto dtype = DataTypeToEnum<T>::value;
+  Tensor forward_tensor(dtype, shape);
+  Tensor swapped_tensor(dtype, shape);
+  std::memcpy((char*)forward_tensor.tensor_data().data(), forward,
+              array_len * bytes_per_elem);
+  std::memcpy((char*)swapped_tensor.tensor_data().data(), swapped,
+              array_len * bytes_per_elem);
+  TF_EXPECT_OK(ByteSwapTensor(&forward_tensor));
+  test::ExpectTensorEqual<T>(forward_tensor, swapped_tensor);
+}
+
+// Unit test of the byte-swapping operations that TensorBundle uses.
+TEST(TensorBundleTest, SwapBytes) {
+  // A bug in the compiler on MacOS causes ByteSwap() and FlipEndiannessBit()
+  // to be removed from the executable if they are only called from templated
+  // functions. As a workaround, we make some dummy calls here.
+  // TODO(frreiss): Remove this workaround when the compiler bug is fixed.
+  ByteSwap(Constant_2x3<int>(42));
+  EXPECT_NE(Status::OK(), FlipEndiannessBit(Prefix("not_a_valid_prefix")));
+
+  // Test patterns, manually swapped so that we aren't relying on the
+  // correctness of our own byte-swapping macros when testing those macros.
+  // At least one of the entries in each list has the sign bit set when
+  // interpreted as a signed int.
+  const int arr_len_16 = 4;
+  const uint16_t forward_16[] = {0x1de5, 0xd017, 0xf1ea, 0xc0a1};
+  const uint16_t swapped_16[] = {0xe51d, 0x17d0, 0xeaf1, 0xa1c0};
+  const int arr_len_32 = 2;
+  const uint32_t forward_32[] = {0x0ddba115, 0xf01dab1e};
+  const uint32_t swapped_32[] = {0x15a1db0d, 0x1eab1df0};
+  const int arr_len_64 = 2;
+  const uint64_t forward_64[] = {0xf005ba11caba1000, 0x5ca1ab1ecab005e5};
+  const uint64_t swapped_64[] = {0x0010baca11ba05f0, 0xe505b0ca1eaba15c};
+
+  // 16-bit types
+  TestByteSwap(forward_16, swapped_16, arr_len_16);
+  TestByteSwap((int16_t*)forward_16, (int16_t*)swapped_16, arr_len_16);
+  TestByteSwap((bfloat16*)forward_16, (bfloat16*)swapped_16, arr_len_16);
```

Comment:
C-style casts are discouraged; use reinterpret_cast

Reply:
Switched to reinterpret_cast